lib/Mir.ml

```diff
@@ -96,18 +257,66 @@ type ('e, 's) prog =
   ; prog_path: string }
 [@@deriving sexp]
 
+let pp_input_vars pp_e ppf {input_vars; _} =
+  pp_io_var_block "input_vars" pp_e ppf input_vars
+
+let pp_output_vars pp_e ppf {output_vars; _} =
+  pp_io_var_block "output_vars" pp_e ppf output_vars
+
+let pp_functions_block pp_s ppf {functions_block; _} =
+  pp_block "functions" pp_s ppf functions_block
+
+let pp_prepare_data pp_s ppf {prepare_data; _} =
+  pp_block "prepare_data" pp_s ppf prepare_data
+
+let pp_prepare_params pp_s ppf {prepare_params; _} =
+  pp_block "prepare_params" pp_s ppf prepare_params
+
+let pp_log_prob pp_s ppf {log_prob; _} = pp_block "log_prob" pp_s ppf log_prob
+
+let pp_generate_quantities pp_s ppf {generate_quantities; _} =
+  pp_block "generate_quantities" pp_s ppf generate_quantities
+
+let pp_transform_inits pp_s ppf {transform_inits; _} =
+  pp_block "transform_inits" pp_s ppf transform_inits
+
+let pp_prog pp_e pp_s ppf prog =
+  Format.open_vbox 0 ;
+  (pp_functions_block pp_s) ppf prog ;
```

Comment:
why this partial application like this?

Reply:
Do you mean why have functions for each element of the `prog` type? If so, this is is to support pretty printing only parts of the MIR.

If you mean partial application in the FP sense, I can't see where I'm doing it in the lines highlighted and it wouldn't have intentionally do this!

Reply:
FP sense. without partial application
```
   (pp_functions_block pp_s) ppf prog ;
```
becomes
```
   pp_functions_block pp_s ppf prog ;
```
unless I'm missing something?

Reply:
Gotcha - I editted from the previous version using the format string and didn't drop the parentheses. I'll remove